solvebio/query.py

```diff
@@ -871,8 +872,9 @@ def join(self, query_b, key, key_b=None, prefix="b_", always_prefix=False):
         # Try to use a unique field for the sub-query data
         query_b_fields = query_b.fields()
 
-        # If a joining key in both datasets is the same then remove it from query_b
-        if key_b == key:
+        # If a joining key in both datasets is the same and
+        # it is not the only one field then remove it from query_b
+        if key_b == key and len(query_b_fields) == 1 and query_b_fields[0].name != key_b:
```

Comment:
What issue does this fix?

Reply:
e.g. https://github.com/solvebio/solvebio-python/blob/bf3394638d5ce39a52e97d6284e893c4744dbfd8/solvebio/test/test_query.py#L378

This will not work if we remove the `gene` from query_b because it is the only field in that query.

Reply:
I updated the link in a previous comment.